react/components/ProductAvailability.tsx

```diff
@@ -24,24 +29,31 @@ function ProductAvailability({
 
   const isLowStock = availableQuantity < threshold
 
-  if (isLowStock && lowStockMessage) {
-    return (
-      <Container>
-        <LowStock
-          text={lowStockMessage}
-          availableQuantity={availableQuantity}
-        />
-      </Container>
-    )
-  }
+  if (!showAvailability) {
```

Comment:
not an impediment, but maybe you can avoid nesting if you invert this condition and use an early return, something like:

```js
...
if(showAvailability){
 return <Container>
      <HighStock text={highStockMessage} />
      <ShowAvailable
        showAvailabilityMessage={showAvailabilityMessage}
        availableQuantity={availableQuantity}
      />
    </Container>
}
...
```
further info: 
https://github.com/Shopify/web-configs/blob/main/packages/eslint-plugin/docs/rules/prefer-early-return.md

The remaining looks good to me